terraform/iam/artifacts.tf

```diff
@@ -0,0 +1,184 @@
+// s3writer for artifacts management
+resource "aws_iam_role" "artifacts-k8s-io-s3writer" {
+  provider = aws.artifacts-k8s-io
+
+  name = "artifacts.k8s.io_s3writer"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          AWS = "arn:aws:iam::768319786644:root"
+        }
+      },
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          Service = "s3.amazonaws.com"
+        }
+      },
+      {
+        Action = "sts:AssumeRole",
+        Effect = "Allow",
+        Principal = {
+          Service = "batchoperations.s3.amazonaws.com"
+        },
+      },
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          AWS = "arn:aws:iam::585803375430:user/artifacts.k8s.io-ci"
+        }
+      },
+      {
+        "Effect" : "Allow",
```

Comment:
This Allow block is needed by the prow jobs, I think.  It'll be the same trusted cluster, but the service-account will be k8s-infra-promoter (at least, that's the one we're using today)